scripts/install_mcljava_with_optimized_gmp.sh

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env sh
+mcl_version="ae1c67a819a81fd645a544d5fb420b980b89816a"
+# exit immediately on error
+set -e
+
+C_TUNING_FLAGS="-march=native -mtune=native"
+
+arch="$(uname -m)"
+
+if [ "$arch" = "arm64" ] || [ "$arch" = "aarch64" ]; then
+  if command -v clang &>/dev/null; then
+    # clang exists and we run on arm64 - clang might be chosen as compiler by CMake, and on this arch, it does not support -march=native
+    # this flag works with both gcc and clang on aarch64, however, and should have the same effect - use it instead
+    C_TUNING_FLAGS="-mcpu=native"
+  fi
+fi
+
+GMP_VERSION=6.2.1
+gmp_from_source(){
+ echo "Cloning GNU gmp."
+ curl https://gmplib.org/download/gmp/gmp-${GMP_VERSION}.tar.xz > gmp-${GMP_VERSION}.tar.xz || (echo "Error downloading GMP" && exit 1)
+ unxz gmp-${GMP_VERSION}.tar.xz || (echo "Could not extract .xz archive - make sure you have unxz installed." && exit 1)
+ tar -xvf gmp-${GMP_VERSION}.tar || (echo "Could not untar the gmp tree." && exit 1)
+ rm gmp-${GMP_VERSION}.tar
+ cd gmp-${GMP_VERSION}
+ ./configure --enable-shared --enable-static --enable-cxx || (echo "Could not determine a GMP configuration" && exit 1)
+ MAKE_PARALLEL=""
+ if command -v nproc &>/dev/null; then
+	 MAKA_PARALLEL="-j $(nproc)"
+ else
+	 echo "nproc was not found on your system - defaulting to default make behavior (usually single-threaded compile)."
+ fi
+ make ${MAKE_PARALLEL} || (echo "Could not make GMP" && exit 1)
+ make check || (echo "Tests for GMP failed!" && exit 1)
+ sudo make install || (echo "Could not install gmp to /usr/" && exit 1)
+ cd ..
+ rm -rf gmp-${GMP_VERSION}
+}
+# check for operating system
+os=""
+if [ "$(uname)" = "Darwin" ]; then
+  os="mac"
+elif [ "$(expr substr $(uname -s) 1 5)" = "Linux" ]; then
+  os="linux"
+elif [ "$(uname)" = "FreeBSD" ]; then
+  os="FreeBSD"
+else
+  echo "Unsupported operating system. This script only works on Linux and macOS."
+  exit 2
+fi
+
+# check that JAVA_INC is given
+if [ $# -eq 0 ]; then
+	echo "Missing Java include argument"
+	echo "Please specify path of your JDK 'include' directory as first argument"
+	if [ "$os" = "linux" ]; then
+    echo "For example: ./install_fast_mcljava_linux_mac.sh /usr/lib/jvm/java-8-openjdk-amd64/include"
```

Comment:
Please rename script in command (here and below)

Reply:
Now uses `$0`.